tower_modules/tower_project/tasks/main.yml

```diff
@@ -63,8 +63,8 @@
     name: "TestProject {{ project_name_rand }}"
     organization: TestOrg1
     scm_type: git
-    scm_url: "https://github.com/ansible/ansible"
-    scm_credential: TestCred1
+    scm_url: https://github.com/ansible/test-playbooks
+    scm_credential: "{{ new_credentials.results[0].id | default('TestCred1') }}"
```

Comment:
Hmmm, I'm not sure we want to default to `TestCred1` here, i'd almost prefer to just see when the id doesnt exist as that might indicate some sort of a bug

Reply:
The non-converted version of the `tower_project ` module requires the name vs ID... I can just remove the `default` bit and have it purely be "ID or Name".

Reply:
Turns out that in order to have `scm_credential` honor the "or" condition, `default()` needs to be included.

Reply:
(which I think is actually OK to make that a default, because the converted module ALSO looks for the name _first_, it just wants an ID if there are identically-named creds)

Reply:
Updated everything to be more simple!